### PR TITLE
feat(price-feed): add RAIRedemptionRatePriceFeed

### PR DESCRIPTION
**Pull request by ashutoshvarma**
**Motivation**
New Off-chain price feed implementation for proposed price feeds in https://github.com/UMAprotocol/UMIPs/pull/232, `R3_10H_TWAP` and `R3_30D_GM`

**Summary**

A new price feed implementation `RAIRedemptionRatePriceFeed` is added which uses subgraphs to retrieve data and calculate both TWAP and Geometric Mean of RAI Redemption Rate Coefficient.

**Details**

Instead of creating two separate implementations for `R3_10H_TWAP` and `R3_30D_GM`, I ended up making a common for both, since both have a common source of data and only post-processing on data is changed.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [x]  Untested

I will be adding the tests shortly

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/financial-templates-lib/src/price-feed/RAIRedemptionRate.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    //         {
+    //           "createdAt": "1617357274",
+    //           "createdAtBlock": "12159386",
+    //           "twentyFourHourlyRate": "0.999613665386634504687653838"
+    //         },
+    //         ...
+    //        ]
+    //     }
+    // }
+    // console.log(response);
```
Comment (chrismaree) [inline packages/financial-templates-lib/src/price-feed/RAIRedemptionRate.js:233]: left over log.

**Comment by ashutoshvarma** [comment]
@chrismaree have you guys come to any consensus regarding price feed with subgraphs? If so then I can continue with tests.

**Comment by chrismaree** [comment]
> @chrismaree have you guys come to any consensus regarding price feed with subgraphs? If so then I can continue with tests.

Hey @ashutoshvarma. We are having a call in the next 30 mins to chat about this and I'll report back on your PR once we've come to decision.

**Comment by chrismaree** [comment]
> > @chrismaree have you guys come to any consensus regarding price feed with subgraphs? If so then I can continue with tests.
> 
> Hey @ashutoshvarma. We are having a call in the next 30 mins to chat about this and I'll report back on your PR once we've come to decision.

@ashutoshvarma we've spoken about this internally and we are happy to use the graph implementation. It is important to us that you provide an open source link to your graph implementation so we can run one on our own nodes. This link to your graph implementation should also be added to your UMIP so any public bot runners can _also_ do the same.

Regarding your PR there are a few additional steps that need to be done:
1) you need to add unit tests for this file.
2) you need to add logic to the `createPriceFeed` which can be found [here](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/CreatePriceFeed.js)
3) you need to add tests to the `createPriceFeed`

Once you've done these steps we'll be able to give you a more detailed review.

Excited to see your product on mainnet!

**Comment by ashutoshvarma** [comment]
> @ashutoshvarma we've spoken about this internally and we are happy to use the graph implementation. It is important to us that you provide an open-source link to your graph implementation so we can run one on our own nodes. This link to your graph implementation should also be added to your UMIP so any public bot runners can _also_ do the same.

This is Great!
The open-source link for the RAI subgraph is already in the UMIP and Sean's [PR](https://github.com/UMAprotocol/UMIPs/pull/274) adds the most recent commit hash as the canonical implementation that voters should reference.

> Regarding your PR there are a few additional steps that need to be done:
> 
> 1. you need to add unit tests for this file.
> 2. you need to add logic to the `createPriceFeed` which can be found [here](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/CreatePriceFeed.js)
> 3. you need to add tests to the `createPriceFeed`

I'm on it

**Comment by chrismaree** [comment]
@ashutoshvarma any updates on this PR? it's been sitting idle for a while.

**Comment by ashutoshvarma** [comment]
@chrismaree Sorry for the delay! A lot going on lately.
WIll try to wrap this soon, hopefully by tomorrow. 
Hope that's not an issue :)

**Comment by daywiss** [comment]
@ashutoshvarma checking in to see how this is coming

**Comment by chrismaree** [comment]
@ashutoshvarma following up with this again :)

**Comment by nicholaspai** [comment]
> @ashutoshvarma following up with this again :)

@ashutoshvarma @chrismaree should we close this for now?

**Comment by chrismaree** [comment]
I think we should close this PR at this point. @UMAprotocol/eng

**Comment by smb2796** [comment]
Closing as this appears to be stale. @ashutoshvarma please feel free to reopen this at a later date if you are going to pick this back up!

**Pull request closed without merging**